ceno_zkvm/src/bin/e2e.rs

```diff
@@ -131,14 +135,77 @@ fn main() {
     );
 
     let max_steps = args.max_steps.unwrap_or(usize::MAX);
-    run_e2e(
-        program,
-        platform,
-        args.stack_size,
-        args.heap_size,
-        hints,
-        max_steps,
+
+    type E = GoldilocksExt2;
+    type Pcs = Basefold<GoldilocksExt2, BasefoldRSParams>;
+
+    let (prover, verifier, zkvm_witness, pi, cycle_num, e2e_start, exit_code) =
+        run_e2e_gen_witness::<E, Pcs>(
+            program,
+            platform,
+            args.stack_size,
+            args.heap_size,
+            hints,
+            max_steps,
+        );
```

Comment:
I see you've moved the high-level logic back to `bin/`. Is this temporary or do you want it to stay this way?

Reply:
Most of the logic stay in src/e2e.rs. bin/e2e.rs will call sub step function.
Here I move 2 part of logic
- timing statistics print out, since this information are across different sub function
- sanity check for verifier with malicious public input. This part more like a sanity check, thus it's not very general to stay in src/e2e.rs. Besides, to make generics type E, PCS work with `panic::catch_unwind` we need to add quite of trait bound. Above 2 reason makes the decision to stay in high level e2e :)

Reply:
Makes sense :+1: